ui/analyse/src/study/studyCtrl.ts

```diff
@@ -404,6 +404,12 @@ export default class StudyCtrl {
 
   instanciateGamebookPlay = () => {
     if (!this.isGamebookPlay()) return (this.gamebookPlay = undefined);
+    // ensure all original nodes have a gamebook entry,
+    // so we can differentiate original nodes from user-made ones
+    treeOps.updateAll(this.ctrl.tree.root, n => {
+      n.gamebook = n.gamebook || {};
+      if (n.shapes) n.gamebook.shapes = n.shapes.slice(0);
+    });
```

Comment:
This needs to take place before the following line or clicking back out of the analysis board causes a null `gamebook` where we expect a gamebook object.